relay-server/src/actors/processor.rs

```diff
@@ -1080,52 +1083,78 @@ impl EnvelopeProcessorService {
     #[cfg(feature = "processing")]
     fn process_profiles(&self, state: &mut ProcessEnvelopeState) {
         state.managed_envelope.retain_items(|item| match item.ty() {
-            ItemType::Profile => match relay_profiling::expand_profile(&item.payload()) {
-                Ok((profile_id, payload)) => {
-                    if payload.len() <= self.config.max_profile_size() {
-                        if let Some(event) = state.event.value_mut() {
-                            if event.ty.value() == Some(&EventType::Transaction) {
-                                let contexts = event.contexts.get_or_insert_with(Contexts::new);
-                                contexts.add(SentryContext::Profile(Box::new(ProfileContext {
-                                    profile_id: Annotated::new(profile_id),
-                                })));
+            ItemType::Profile => {
+                let tags: BTreeMap<String, String> = match state.event.value() {
+                    Some(event) => {
+                        let mut tags = extract_universal_tags(
+                            event,
+                            &TransactionMetricsConfig {
+                                version: 1,
+                                extract_custom_tags: BTreeSet::from(["device.class".to_string()]),
+                                ..Default::default()
+                            },
+                        );
+
+                        // Overwrite the transaction name to keep the original one
+                        tags.insert(
+                            "transaction".to_owned(),
+                            event.get_transaction_source().to_string(),
```

Comment:
Don't you want `event.transaction.value()` here?

Reply:
Indeed.